src/bin/cargo/cli.rs

```diff
@@ -55,6 +55,13 @@ pub fn main(config: &mut Config) -> CliResult {
         }
     };
 
+    if let Some(new_cwd) = args.get_one::<std::path::PathBuf>("directory") {
+        // Change the configured directory to work in, before any config files are read
+        config
+            .set_cwd(new_cwd.as_path())
```

Comment:
From PR

> Passing a path with a trailing '/' will result in failure. It is unclear whether this is a result of improper input sanitization, or whether the config.cwd value is being improperly handled on output. In either case this needs to be resolved before this PR is merge-ready.

(to ensure we track this blocker)

Reply:
This issue appears to have been a side effect of corruption of my local rustup toolchain.  Soon thereafter the issue started appearing regardless of path formatting, command line arguments, etc. After a forced rustup, the issue disappeared entirely.